gems/activerecord/6.0/activerecord.rbs

```diff
@@ -51,29 +51,31 @@ module ActiveRecord
     def self.validates: (*untyped) -> void
 
     # callbacks
-    def self.after_commit: (*untyped) ?{ () -> untyped} -> void
-    def self.after_create_commit: (*untyped) -> void | ...
-    def self.after_update_commit: (*untyped) -> void | ...
-    def self.after_destroy_commit: (*untyped) -> void | ...
-    def self.after_save_commit: (*untyped) -> void | ...
-    def self.after_create: (*untyped) ?{ () -> untyped} -> void
-    def self.after_destroy: (*untyped) ?{ () -> untyped} -> void
-    def self.after_rollback: (*untyped) ?{ () -> untyped} -> void
-    def self.after_save: (*untyped) ?{ () -> untyped} -> void
-    def self.after_update: (*untyped) ?{ () -> untyped} -> void
-    def self.after_validation: (*untyped) ?{ () -> untyped} -> void
-    def self.after_initialize: (*untyped) ?{ () -> untyped} -> void
-    def self.after_find: (*untyped) ?{ () -> untyped} -> void
-    def self.after_touch: (*untyped) ?{ () -> untyped} -> void
-    def self.around_create: (*untyped) ?{ () -> untyped} -> void
-    def self.around_destroy: (*untyped) ?{ () -> untyped} -> void
-    def self.around_save: (*untyped) ?{ () -> untyped} -> void
-    def self.around_update: (*untyped) ?{ () -> untyped} -> void
-    def self.before_create: (*untyped) ?{ () -> untyped} -> void
-    def self.before_destroy: (*untyped) ?{ () -> untyped} -> void
-    def self.before_save: (*untyped) ?{ () -> untyped} -> void
-    def self.before_update: (*untyped) ?{ () -> untyped} -> void
-    def self.before_validation: (*untyped) ?{ () -> untyped} -> void
+    type callback = Symbol | ^() [self: instance] -> void
+    type condition = Symbol | ^() [self: instance] -> bool
+    def self.after_commit: (*callback, ?if: condition, ?unless: condition, **untyped) ?{ () [self: instance] -> untyped } -> void
```

Comment:
Today, I added types for `if:` and `unless:` keyword arguments. This is needed to support models having conditional hooks.